packages/tools/src/index.ts

```diff
@@ -0,0 +1,179 @@
+// @ts-ignore
+import AsBind from "as-bind/dist/as-bind.cjs.js";
+import fs from "fs";
+
+type ToolParameters = {
+  type: string | "object";
+  properties: Record<string, { type: string; description?: string }>;
+  required?: string[];
+};
+
+interface ToolMetadata {
+  description: string;
+  name: string;
+  parameters?: ToolParameters;
+  argsKwargs?: Record<string, any>;
```

Comment:
`argsKwargs` is currently not used by the agent - I would remove this complexity from the tools package for the start